ee/modules/030-cloud-provider-openstack/hooks/migrate/migrate_terraform_state.go

```diff
@@ -0,0 +1,194 @@
+)
+
+const TerraformStateDataKey = "node-tf-state.json"
+const TerraformStateNamespace = "d8-system"
+const OpenstackV2ResourceType = "openstack_blockstorage_volume_v2"
+const OpenstackV3ResourceType = "openstack_blockstorage_volume_v3"
+
+var _ = sdk.RegisterFunc(&go_hook.HookConfig{
+	OnStartup: &go_hook.OrderedConfig{Order: 10},
+}, dependency.WithExternalDependencies(openstackTerraformStateMigration))
+
+func openstackTerraformStateMigration(input *go_hook.HookInput, dc dependency.Container) error {
+	input.LogEntry.Infof("Starting Terraform migrate hook")
+	kubeCl, err := dc.GetK8sClient()
+	if err != nil {
+		return fmt.Errorf("could not initialize Kubernetes client: %v", err)
+	}
+
+	terraformStateLabels := map[string]string{
+		"node.deckhouse.io/terraform-state": "",
+	}
+
+	terraformStateSecrets, err := kubeCl.CoreV1().
+		Secrets(TerraformStateNamespace).
+		List(context.TODO(), metav1.ListOptions{LabelSelector: metav1.FormatLabelSelector(metav1.SetAsLabelSelector(terraformStateLabels))})
+	if err != nil {
+		return fmt.Errorf("failed to get Terraform state secrets in namespace %s with labels %s. The migration process has been aborted", TerraformStateNamespace, terraformStateLabels)
+	}
+
+	if terraformStateSecrets.Items == nil {
+		input.LogEntry.Infof("Terraform state not found. Migration is not needed.")
+		return nil
+	}
+
+	for _, secret := range terraformStateSecrets.Items {
+		input.LogEntry.Infof("Proceeding with Secret/%s/%s", TerraformStateNamespace, secret.ObjectMeta.Name)
+		backupSecretName := secret.ObjectMeta.Name + "-backup"
+
+		secretBackupExists, err := SecretBackupExists(backupSecretName, TerraformStateNamespace, kubeCl, input)
+		if secretBackupExists && err == nil {
+			input.LogEntry.Infof("Secret backup with name %s/%s already exists! Migration is not needed for Secret/%s/%s", TerraformStateNamespace, backupSecretName, TerraformStateNamespace, secret.ObjectMeta.Name)
+			continue
+		}
+		input.LogEntry.Infof("Secret backup with name %s/%s doesn't exist! Continue migration for Secret/%s/%s", TerraformStateNamespace, backupSecretName, TerraformStateNamespace, secret.ObjectMeta.Name)
+		terraformStateRaw, ok := secret.Data[TerraformStateDataKey]
+		if !ok {
+			return fmt.Errorf("key %s not found in Secret/%s/%s. ", TerraformStateDataKey, TerraformStateNamespace, secret.ObjectMeta.Name)
+		}
+
+		if !gjson.ValidBytes(terraformStateRaw) {
+			return fmt.Errorf("invalid JSON: %s", terraformStateRaw)
+		}
+
+		terraformState := gjson.ParseBytes(terraformStateRaw)
+		openstackV2Resources := terraformState.Get("resources.#(type==\"openstack_blockstorage_volume_v2\")#")
+
+		if len(openstackV2Resources.Array()) == 0 {
+			input.LogEntry.Infof("No old resources found. Migration is not needed for Secret/%s/%s.", TerraformStateNamespace, secret.ObjectMeta.Name)
+			continue
+		}
+
+		input.LogEntry.Infof("Found old resources. Starting migration for Secret/%s/%s", TerraformStateNamespace, secret.ObjectMeta.Name)
```

Comment:
unnecessary

Reply:
I've removed this line: `input.LogEntry.Infof("Found old resources. Starting migration for Secret/%s/%s", TerraformStateNamespace, secret.ObjectMeta.Name)`